src/shared/dropdown-list-v2/nested-dropdown-list.tsx

```diff
@@ -0,0 +1,566 @@
+            const toggleable = hasSubItems && !selectableCategory;
+
+            return (
+                <ListItemContainer
+                    key={`[${keyPath.join("---")}]`}
+                    $visible={visible}
+                >
+                    {maxLevel > 0 && <Indent $level={level} />}
+                    {maxLevel > 0 && !hasSubItems && multiSelect && (
+                        <UnexpandableIndicator />
+                    )}
+                    <ListItem
+                        aria-checked={checked}
+                        aria-expanded={hasSubItems ? expanded : undefined}
+                        aria-level={level + 1}
+                        aria-posinset={indexInParent + 1}
+                        aria-setsize={parentSetSize}
+                        data-testid="list-item"
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            if (toggleable) {
+                                toggleCategory(i, !expanded);
+                            } else {
+                                handleListItemClick(i, keyPath);
+                            }
+                        }}
+                        onMouseEnter={() => handleListItemHover(i)}
+                        ref={(node) => (listItemRefs.current[i] = node)}
+                        role="treeitem"
+                        tabIndex={active ? 0 : -1}
+                        $active={active}
+                        $toggleable={toggleable}
+                    >
+                        {hasSubItems && (
+                            // not an actual button, only required for visual display
+                            <ExpandButton
+                                data-testid="toggle-category-button"
+                                onClick={(e) => {
+                                    e.stopPropagation();
+                                    toggleCategory(i, !expanded);
+                                }}
+                                $expanded={expanded}
+                            >
+                                <CaretRightIcon />
+                            </ExpandButton>
+                        )}
+                        {renderSelectionIcon(listItem)}
+                        <DropdownLabel
+                            bold={hasSubItems}
+                            label={item.label}
+                            selected={!!checked}
+                            truncationType={itemTruncationType}
+                            maxLines={itemMaxLines}
+                        ></DropdownLabel>
+                    </ListItem>
+                </ListItemContainer>
+            );
+        });
+    };
+
+    const renderList = () => {
+        return (
+            <List>
+                {renderSearchInput()}
+                {renderSelectAll()}
+                {renderNoResults()}
+                {renderLoading()}
+                {renderTryAgain()}
```

Comment:
thanks for the suggestion, this could be extracted to a `DropdownResult` component that handles the loading/success/fail display states

will refactor this in a separate PR